bundles/org.openhab.voice.whisperstt/src/main/java/org/openhab/voice/whisperstt/internal/WhisperConfigOptionProvider.java

```diff
@@ -0,0 +1,77 @@
+/**
+ * Copyright (c) 2010-2024 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.voice.whisperstt.internal;
+
+import static org.openhab.voice.whisperstt.internal.WhisperSTTConstants.SERVICE_CATEGORY;
+import static org.openhab.voice.whisperstt.internal.WhisperSTTConstants.SERVICE_ID;
+import static org.openhab.voice.whisperstt.internal.WhisperSTTConstants.SERVICE_NAME;
+import static org.openhab.voice.whisperstt.internal.WhisperSTTConstants.SERVICE_PID;
+import static org.openhab.voice.whisperstt.internal.WhisperSTTService.WHISPER_FOLDER;
+
+import java.net.URI;
+import java.util.Collection;
+import java.util.List;
+import java.util.Locale;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.core.config.core.ConfigOptionProvider;
+import org.openhab.core.config.core.ConfigurableService;
+import org.openhab.core.config.core.ParameterOption;
+import org.osgi.framework.Constants;
+import org.osgi.service.component.annotations.Component;
+
+/**
+ * The {@link WhisperConfigOptionProvider} class provides some dynamic configuration options
+ *
+ * @author Miguel Álvarez - Initial contribution
+ */
+@Component(service = ConfigOptionProvider.class, configurationPid = SERVICE_PID, property = Constants.SERVICE_PID + "="
+        + SERVICE_PID)
+@ConfigurableService(category = SERVICE_CATEGORY, label = SERVICE_NAME, description_uri = SERVICE_CATEGORY + ":"
```

Comment:
For consistency with other voice services, prefer this as label: SERVICE_NAME + " Speech-to-Text"